aip/aog/3010.md

```diff
@@ -0,0 +1,60 @@
+---
+aip:
+  id: 3010
+  state: draft
+  created: 2020-01-14
+  updated: 2020-01-14
+  type: guidance
+  scope: aog
+js:
+  - /assets/js/graphviz/viz.js
+  - /assets/js/graphviz/lite.render.js
+  - /assets/js/aip/aip-graphviz.js
+permalink: /aog/3010
+---
+
+# Actions on Google Vertical Integration Webhook Format
+
+If an action is part of a vertical program that requires a webhook for Google
+to integrate with the action, the webhook format **must** follow this AIP.
+
+## Guidance
+
+### Custom method name
+
+Custom methods in a vertical program are defined by Google and implemented by
+third-party actions. Guidelines related to names of custom methods in
+[AIP-136](../0136.md) **should** be applied.
+
+- The name of the method **should** be a verb followed by a noun.
+- The name must not contain prepositions (“for”, “with”, etc.).
+- If word separation is required for the method name, `lowerCamelCase`
+  **should** be used.
+
+Examples:
+
+- `getAccountBalance`
+- `completeTransaction`
+
+### Base URL in Action Package
+
+The base URL configured in Action Package defines the common part of the actual
+execution URL of the webhook. The base URL **may** include a trailing slash,
+but **must** work if `:customMethodName` is appended directly. The endpoint
+**must** be HTTPS and **must** have a valid certificate.
+
+Examples:
+
+- `https://us-central1.cloudfunctions.net/myWebhook/` (**not**
+  `https://us-central1.cloudfunctions.net/myWebhook`)
+- `https://test.com/assistant/api`
+
```

Comment:
I think you need to a small explanation here why the second case fails, because some readers will be unfamiliar with cloud functions. Maybe something instead of "Examples:" like "For example, Google Cloud Function base URLs without a trailing slash do not work, because the first segment of the path is used to identify the webhook:"

Reply:
good suggestion! added